[36269] Add "other" to pluralization rules for one

oliverguenther opened the pull request:
For languages that do not have "one" as a pluralization rule (japanese, for example), the pluralization rule defined by I18n.js does not return a valid string for count: 1.

Since we probably do not want to define custom pluralization rules for all languages, we can fall back to the "other" key if "one" is not defined.

https://community.openproject.org/wp/36269

oliverguenther committed:
[36269] Add "other" to pluralization rules for one

For languages that do not have "one" as a pluralization rule (japanese,
for example), the pluralization rule defined by I18n.js does not return
a valid string for count: 1.

Since we probably do not want to define custom pluralization rules for
all languages, we can fall back to the "other" key if "one" is not
defined.

https://community.openproject.com/wp/36269

ulferts commented on frontend/src/app/init-locale.ts:
```diff
@@ -0,0 +1,50 @@
+      case 1:
+        return ["one", "other"];
+      default:
+        return ["other"];
```
There seem to be more cases (e.g. few in Russian https://stackoverflow.com/questions/6166064/i18n-pluralization) which are not handled here. Did you look into the overridden method to check if anything might get broken by us overwriting the function?

oliverguenther replied:
I18n.js ships with only the english version: 

https://github.com/opf/openproject/blob/dev/frontend/src/vendor/i18n.js#L367-L373

So extending it with a fallback for "other" from "one" will not break anything that wasn't broken before.

It does not really handle special cases such as russian. For that, we would need to define our own pluralization rules, as the project does not have them. Might be better to switch to another I18n provider for the frontend instead

ulferts replied:
Thanks for looking into it, @oliverguenther

oliverguenther committed:
Explicitly wait before clicking on viewpoint

The click happens before the side tree is rendered by xeokit, and not being updated afterwards
